src/Associations.cc

```diff
@@ -265,40 +262,53 @@ void Associations::associateRefStars(double matchCutInArcSec, const Gtransfo *gt
 
 void Associations::selectFittedStars(int minMeasurements) {
     LOGLS_INFO(_log, "Fitted stars before measurement # cut: " << fittedStarList.size());
-    /* first pass : remove objects that have less than a
-       certain number of measurements.
-    */
+
+    // Clear positions to make them the average of the measuredStars.
+    for (auto &fittedStar : fittedStarList) {
+        fittedStar->x = 0.0;
+        fittedStar->y = 0.0;
+    }
+
+    // first pass: remove objects that have less than a certain number of measurements.
     for (auto const &ccdImage : ccdImageList) {
+        const Gtransfo *toCommonTangentPlane = ccdImage->getPix2CommonTangentPlane();
         MeasuredStarList &catalog = ccdImage->getCatalogForFit();
+        // Iteration happens internal to the loop, as we may delete measuredStars from catalog.
         for (MeasuredStarIterator mi = catalog.begin(); mi != catalog.end();) {
             MeasuredStar &mstar = **mi;
 
             auto fstar = mstar.getFittedStar();
+            // measuredStar has no fittedStar: move on.
             if (!fstar) {
                 ++mi;
                 continue;
             }
 
-            /*  keep FittedStar's which either have a minimum number of
-                measurements, or are matched to a RefStar
-            */
+            // keep FittedStars which either have a minimum number of
+            // measurements, or are matched to a RefStar
             if (!fstar->getRefStar() && fstar->getMeasurementCount() < minMeasurements) {
                 auto f = std::const_pointer_cast<FittedStar>(fstar);
                 f->getMeasurementCount()--;
-                mi = catalog.erase(mi);
-            } else
+                mi = catalog.erase(mi);  // mi now points to the next measuredStar.
+            } else {
+                auto f = std::const_pointer_cast<FittedStar>(fstar);
+                auto point = toCommonTangentPlane->apply(mstar);
+                f->x += point.x;
+                f->y += point.y;
                 ++mi;
+            }
         }  // end loop on objects in catalog
     }      // end loop on catalogs
 
-    /* now FittedStars with less than minMeasurements should have
-       zero measurementCount; */
-
+    // now FittedStars with less than minMeasurements should have zero measurementCount.
     for (FittedStarIterator fi = fittedStarList.begin(); fi != fittedStarList.end();) {
-        if ((*fi)->getMeasurementCount() == 0)
+        if ((*fi)->getMeasurementCount() == 0) {
             fi = fittedStarList.erase(fi);
-        else
+        } else {
+            (*fi)->x /= (*fi)->getMeasurementCount();
+            (*fi)->y /= (*fi)->getMeasurementCount();
```

Comment:
Ok, now I think I see what you're trying to do, but it's still very misleading to see this in the middle of what should be a filtering method.